### ISSUE TITLE
Specify destination folder during uploads

### ISSUE THREAD

**Issue by natrow** [typst/webapp-issues#172]
### Description

Currently uploaded files (usually?) go to the root directory of the project. It's very tedious to upload and then move large amounts of files to the target folder. While #63 partially solves the problem, it would still be helpful to choose the target during file upload.

I say usually because sometimes files seem to be placed in a destination but it depends on how they're uploaded (e.g. drag and drop).

### Use Case

Convenience feature for people who upload large amounts of files at once and/or sort their files. One use-case is for those who upload similarly-named files into different directories to organize them.

**Comment by Stern1710** [comment]
I very much support this. Currently, every time I upload a new image or something alike, I have to manually sort it into the data structure. I sadly cannot always just upload a folder with everything as I need it, so that would be highly appreciated.

**Comment by laurmaedje** [comment]
The best way at the moment is to just drag-and-drop the file into the appropriate location in the file panel instead of using the upload button. But something in the upload modal itself could be good as well.

**Comment by Stern1710** [comment]
Thank you for the reply. I had not considered that this is possible, as I had never tried drag&drop inside the tree from an external source. Thanks, that made it easier. However, I feel like this could be either documented somewhere (or I missed it, sorry) or at least mentioned during the file upload that this is also a possible way.

**Comment by laurmaedje** [comment]
Yeah, web app docs are unfortunately practically non-existent at the moment. :/

**Comment by Stern1710** [comment]
It probably sounds stupid, but any way one could help with that?

**Comment by laurmaedje** [comment]
@Stern1710 I think helping out with the writing itself would be rather hard as we'd first need to come up with a good structure and general idea and that will need quite a bit of internal discussion. But if there are things you'd like to see or specific ideas you have, that we can base such internal discussion on, that could be helpful!
